carmin-server/server/resources/register.py

```diff
@@ -1,26 +1,40 @@
 from server import db
 from flask_restful import Resource
+from sqlalchemy.exc import IntegrityError
 from .models.authentication_credentials import AuthenticationCredentials, AuthenticationCredentialsSchema
 from server.common.error_codes_and_messages import USERNAME_ALREADY_EXISTS
 from server.database.models.user import User, Role
-from .decorators import admin_only, marshal_request, marshal_response
+from .decorators import admin_only, unmarshal_request, marshal_response, get_db_session
+from server.resources.helpers.register import create_user_directory, already_existing_username_error_code_and_message
+from server.resources.models.error_code_and_message import ErrorCodeAndMessage
+from server.common.error_codes_and_messages import UNEXPECTED_ERROR
 
 
 class Register(Resource):
     @admin_only
-    @marshal_request(AuthenticationCredentialsSchema())
+    @get_db_session
```

Comment:
Discussed and agreed to keep it for the moment. In case we feel like needing extra decorators in the future, we will reconsider its importance further and possibly move it to a utils fonction.